### PR TITLE
Only perform certain actions with managed storage if the VM is s…

### PR DESCRIPTION
**Pull request by mike-tutkowski**
…tarted up in a new cluster (not if it is just started up on a new host in the same cluster).

## Description
Example: A VM that uses managed storage is stopped. The VM is then started on a different host in the same cluster. The Start operation fails.

To get around this issue, you must either start the VM up on the same host or on a host in a different cluster.

The reason is due to a slightly erroneous check in VolumeOrchestrator.prepare.

To solve this issue, we should be checking if the cluster ID changes, not if the host ID changes.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?
I noticed that I could not start a stopped VM that uses managed storage on a different host in the same cluster as the host it was originally running on.

After I changed the code (this PR), I re-tested the scenario. I can now start the VM up on a different host in the same cluster as the host it was originally running on. Also, I can still start it up on the same host as it was originally running on or on a host in a different cluster.

## Checklist:
- [x] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

@blueorangutan package

### CODE REVIEW

**Comment by mike-tutkowski** [comment]
I plan to write a Marvin test for this, but - in the meanwhile - wanted to get this PR opened so reviewers could provide comments on the production-focused code.

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by mike-tutkowski** [comment]
I doubt this Jenkins error has anything to do with this PR:

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running streamer.ByteBufferTest
Tests run: 400, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.082 sec - in streamer.ByteBufferTest
Running streamer.BaseElementTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.116 sec - in streamer.BaseElementTest
Running common.ClientTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.423 sec - in common.ClientTest
Running rdpclient.MockServerTest
Error in mock server: Received fatal alert: handshake_failure
javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure
	at sun.security.ssl.Alerts.getSSLException(Alerts.java:192)
	at sun.security.ssl.Alerts.getSSLException(Alerts.java:154)
	at sun.security.ssl.SSLSocketImpl.recvAlert(SSLSocketImpl.java:2038)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1135)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1385)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1413)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1397)
	at streamer.debug.MockServer.run(MockServer.java:122)
	at java.lang.Thread.run(Thread.java:748)
Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.461 sec <<< FAILURE! - in rdpclient.MockServerTest
testIsMockServerCanUpgradeConnectionToSsl(rdpclient.MockServerTest)  Time elapsed: 0.456 sec  <<< ERROR!
javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)
	at sun.security.ssl.Handshaker.activate(Handshaker.java:529)
	at sun.security.ssl.SSLSocketImpl.kickstartHandshake(SSLSocketImpl.java:1492)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1361)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1413)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1397)
	at rdpclient.MockServerTest.testIsMockServerCanUpgradeConnectionToSsl(MockServerTest.java:166)


Results :

Tests in error: 
  MockServerTest.testIsMockServerCanUpgradeConnectionToSsl:166 » SSLHandshake No...

File: engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java [hunk 1]
```diff
@@ -1349,30 +1349,29 @@ public void prepare(VirtualMachineProfile vm, DeployDestination dest) throws Sto
...
+                    Host host = _hostDao.findById(vm.getVirtualMachine().getHostId());
 
-                    if (oldHostId != hostId) {
-                        Host oldHost = _hostDao.findById(oldHostId);
-                        DataStore storagePool = dataStoreMgr.getDataStore(pool.getId(), DataStoreRole.Primary);
+                    long lastClusterId = lastHost == null || lastHost.getClusterId() == null ? -1 : lastHost.getClusterId();
+                    long clusterId = host == null || host.getClusterId() == null ? -1 : host.getClusterId();
 
-                        storageMgr.removeStoragePoolFromCluster(oldHostId, vol.get_iScsiName(), pool);
+                    if (lastClusterId != clusterId) {
```
Comment (nitin-maharana) [inline engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java]: @mike-tutkowski, I doubt if we need to handle the else part where the target cluster is same as the old one. Where we can put a warning/error that this action cannot be performed on the same cluster different host etc.. Thanks.

**Comment by mike-tutkowski** [comment]
Going to open a new PR for this for 4.11.1.

**Pull request closed without merging**